JavaScript/JavaScript-Classes.md

```diff
@@ -0,0 +1,171 @@
+#JavaScript Functional Classes
+
+###Introduction
+
+JavaScript is a prototypal object-oriented language, meaning that objects can inherit directly from other objects.  This is a bit different to those who have learned to develop in other langauges like Java or C# where a subclass will inherit from a superclass, and objects are instances of classes. 
+
+If you want to create multiple objects with similar methods and properties in JavaScript, the best options are functional classes, or the constructor function. There are four styles of functional classes.
+
+ * Functional
+
+ * Functional-Shared
+
+ * Prototypal
+
+ * Pseudoclassical
+
+Each of these four is a different approach to object construction; they all accomplish the same thing in a slightly different way.
+
+Take for example, a street, which has many houses. Every house on the street requires a colour for the house, a front door with methods to open and close the door. A very simple example of where houses may be thought of as objects. 
+
+We can implement functional classes here, as every instance created by the function will have the attributes of house colour and the methods to open and close the door.  
+
+From here we can potentially create hundreds, thousands or even more new house objects, which are similar.  So each functional class performs the same basic steps to generate the new houses.
+
+* generate an object
+* assign properties
+* add methods
+* return that object
+
+Each of the four styles has its pros and cons.
+
+####Functional
+Firstly **functional style** is **fast, very simple and clear in object creation**.  However all methods will be duplicated for every single object created, so a new function is stored in memory every time the object is generated.
```

Comment:
Suggest changing:
>However all methods will be duplicated for every single object created, and a new function is stored in memory every time the object is generated. If you are working with large numbers of objects, this may not be the best choice.